### ISSUE TITLE
reference mismatch error

### ISSUE THREAD

**Issue by DavidEGray** [SciLifeLab/TIDDIT#43]
Hi.
TIDDIT is running fine for me on GRCh37 bams, but I'm running into a problem with GRCh38.

I'm getting a reference mismatch error, here are the last few lines of output and the error:
working on seqence HLA-DRB1*08:03:02
working on seqence HLA-DRB1*09:21
working on seqence HLA-DRB1*11:01:02
working on seqence HLA-DRB1*15:02:01
variant calling time consumption= 5609s
error: reference mismatch!

I checked to make sure that the bam is aligned to the reference fasta I'm specifying, and I checked to make sure that the chromosome names are the same in both.  The only difference I can see is unaligned reads with chromosome "*" in the bam.

Any ideas?

Thanks!

**Comment by szilvajuhos** [comment]
Hullo, likely HLA naming convention screws up things, HLA names contain stars like HLA-DRB1\*08:03:02, HLA-DRB1\*09:21 etc. IMHO it would be better to exclude all the HLA sequences altogether as they are relatively short (3-15K long) homologous references.

**Comment by szilvajuhos** [comment]
I would ask @J35P312 to have a final word on that, my first naiive approach would be to keep HLA for alignment, and ignore reads mapped to these sequences (i.e. filter out from BAM). The MHC structure itself is very interesting, but including all these short references for structural variants maybe is an overkill.

**Comment by J35P312** [comment]
Good evening!
How are the rest of the chromosomes named? Do they have the "chr" prefix (ie chr1,chr2...) or not?
You might have to change the -s parameter of TIDDIT. The -s parameter is a list of chromosomes used as a reference for finding the ploidy of the remaining chromosomes. 
The names of the chromosomes of the -s list must also match your reference, for example:

-s chr1,chr2,chr3,chr4,chr5
or
-s 1,2,3,4,5
depending on the naming of the chromosomes in the reference fasta. Per default, tiddit will use the following chromosomes as a reference:

1,15,14,9,2

If the chromosomes or your reference are not named like that, TIDDIT will produce the "reference mismatch error". I have made that  error message a bit more user friendly since version 2.0.5 =P

I have never run TIDDIT on a bam file containing HLA contigs, but I will try it tomorrow and see what happens.
I will keep you updated!

**Comment by DavidEGray** [comment]
The -s option worked, thank you.

The reference does have the "chr" prefix, so I added -s chr1,chr15,chr14,chr9,chr2 and that cleared it up.

Just for the record, I also tried using a bam that had only the usual 22 numbered chromosomes and X and Y, ie no oddly named HLA, alternate, or decoy chromosomes, and that did not fix it.  When I ran it with the -s option and it worked, I was using the full bam with all the weird chromosomes, so the odd HLA naming conventions in hg38 are not the cause of the problem.

Thanks again for the help!
